tracing-subscriber/src/filter/env/field.rs

```diff
@@ -36,14 +36,23 @@ pub(crate) struct MatchVisitor<'a> {
     inner: &'a SpanMatch,
 }
 
-#[derive(Debug, Clone, PartialOrd, Ord, Eq, PartialEq)]
+#[derive(Debug, Clone, PartialOrd, PartialEq)]
```

Comment:
the manual implementation of `Ord` for a type that derives `PartialOrd` triggers a clippy lint: https://github.com/tokio-rs/tracing/runs/3363637695#step:4:753

I think we need to add a manual `PartialOrd` impl as well, to fix the clippy lint.

Reply:
:+1: